staging/src/k8s.io/sample-apiserver/pkg/registry/registry.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+
+// rest implements a RESTStorage for API services against etcd
+type REST struct {
```

Comment:
This is the spot I expected `RESTInPeace(rest.StandardStorage, error) rest.StandardStorage` to be here.  Then each individual registry still has a `NewREST(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter) *registry.REST)` which can be wrapped.